plugins/trivy/cmd/trivy/main.go

```diff
@@ -100,4 +118,43 @@ func main() {
 	if err != nil {
 		panic(err)
 	}
+	logrus.Info("Finished writing file ", outputFile)
+}
+
+func getNewestVersionsToScan(ctx context.Context, imageWithVulns []models.ImageReport) []models.Image {
+	versionsChan := make(chan newestVersions, len(imageWithVulns))
+	for _, i := range imageWithVulns {
+		go getNewestVersions(versionsChan, ctx, i)
+	}
+	newImagesToScan := []models.Image{}
+	for i := 0; i < len(imageWithVulns); i++ {
+		vc := <-versionsChan
+		if vc.err == nil {
+			for _, v := range vc.versions {
+				newImagesToScan = append(newImagesToScan, models.Image{
+					ID:                 fmt.Sprintf("%v:%v", vc.repo, v),
+					Name:               fmt.Sprintf("%v:%v", vc.repo, v),
+					PullRef:            fmt.Sprintf("%v:%v", vc.repo, v),
+					RecommendationOnly: true,
+				})
+			}
+		}
+	}
+	return newImagesToScan
+}
+
+func getNewestVersions(versionsChan chan newestVersions, ctx context.Context, img models.ImageReport) {
+	repo := strings.Split(img.Name, ":")[0]
+	tag := strings.Split(img.Name, ":")[1]
```

Comment:
Would be good to check that `img.Name` is the right format so this doesn't crash. E.g.

```suggestion
  parts := strings.Split(img.Name, ":")
  if len(parts) != 2 {
    return
  }
  repo := parts[0]
  tag := parts[1]
```

Reply:
Good point